File: backend/app/schemas/agent_commission.py
from datetime import date, datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AgentCommissionCreate(BaseModel):
    company_id: UUID | None = None
    agent_id: UUID
    bill_no: str | None = None
    supplier_name: str | None = None
    vehicle_no: str | None = None
    bill_total: Decimal
    commission_pct: Decimal
    commission_amount: Decimal
    payment_date: date | None = None
    paid: bool = False


class AgentCommissionUpdate(BaseModel):
    company_id: UUID | None = None
    agent_id: UUID | None = None
    bill_no: str | None = None
    supplier_name: str | None = None
    vehicle_no: str | None = None
    bill_total: Decimal | None = None
    commission_pct: Decimal | None = None
    commission_amount: Decimal | None = None
    payment_date: date | None = None
    paid: bool | None = None


class AgentCommissionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    company_id: UUID | None
    agent_id: UUID
    bill_no: str | None
    supplier_name: str | None
    vehicle_no: str | None
    bill_total: Decimal
    commission_pct: Decimal
    commission_amount: Decimal
    payment_date: date | None
    paid: bool
    created_at: datetime
    updated_at: datetime

File: backend-spring/src/main/java/com/nearkart/entity/AgentCommission.java
package com.nearkart.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "agent_commissions")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class AgentCommission {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "company_id")
    private UUID companyId;

    @Column(name = "agent_id", nullable = false)
    private UUID agentId;

    @Column(name = "bill_no")
    private String billNo;

    @Column(name = "supplier_name")
    private String supplierName;

    @Column(name = "vehicle_no")
    private String vehicleNo;

    @Column(name = "bill_total", nullable = false, precision = 14, scale = 2)
    private BigDecimal billTotal;

    @Column(name = "commission_pct", nullable = false, precision = 5, scale = 2)
    private BigDecimal commissionPct;

    @Column(name = "commission_amount", nullable = false, precision = 14, scale = 2)
    private BigDecimal commissionAmount;

    @Column(name = "payment_date")
    private LocalDate paymentDate;

    @Builder.Default
    private Boolean paid = false;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "agent_id", insertable = false, updatable = false)
    private Agent agent;

    @PrePersist
    protected void onCreate() { OffsetDateTime now = OffsetDateTime.now(); createdAt = now; updatedAt = now; }

    @PreUpdate
    protected void onUpdate() { updatedAt = OffsetDateTime.now(); }
}
